--- a/src/core/linux/SDL_evdev.c	2016-10-20 05:50:33.000000000 +0200
+++ b/src/core/linux/SDL_evdev.c	2021-05-16 09:03:27.582541612 +0200
@@ -158,14 +158,12 @@
     if (ioctl(tty_fd, KDGKBMODE, old_kb_mode) < 0) {
         return SDL_SetError("Failed to get keyboard mode during muting");
     }
-    
-    /* FIXME: atm this absolutely ruins the vt, and KDSKBMUTE isn't implemented
-       in the kernel */
-    /*
     if (ioctl(tty_fd, KDSKBMODE, K_OFF) < 0) {
         return SDL_SetError("Failed to set keyboard mode during muting");
     }
-    */
+	if ( ioctl(tty_fd, KDSETMODE, KD_GRAPHICS) < 0 ) {
+        return SDL_SetError("EVDEV: Failed to enable graphics mode");
+    }
     
     return 0;  
 }
@@ -173,12 +171,12 @@
 /* Restore the keyboard mode for given tty */
 static void SDL_EVDEV_unmute_keyboard(int tty_fd, int kb_mode)
 {
-    /* read above */
-    /*
     if (ioctl(tty_fd, KDSKBMODE, kb_mode) < 0) {
         SDL_Log("Failed to unmute keyboard");
     }
-    */
+	if ( ioctl(tty_fd, KDSETMODE, KD_TEXT) ) {
+        SDL_Log("EVDEV: Failed to restore text mode");
+    }
 }
 
 static int SDL_EVDEV_get_active_tty()
