diff --git a/src/gallium/drivers/etna/etna_rasterizer.h b/src/gallium/drivers/etna/etna_rasterizer.h
index 0574391..3744fdc 100644
--- a/src/gallium/drivers/etna/etna_rasterizer.h
+++ b/src/gallium/drivers/etna/etna_rasterizer.h
@@ -28,5 +28,9 @@ struct pipe_context;
 
 void etna_pipe_rasterizer_init(struct pipe_context *pipe);
 
+#define VIVS_PA_SYSTEM_MODE_UNK0  0x00000001
+#define VIVS_PA_SYSTEM_MODE_UNK4  0x00000010
+#define VIVS_PA_SYSTEM_MODE       0x00000a28
+
 #endif
 
diff --git a/src/gallium/drivers/etna/etna_zsa.h b/src/gallium/drivers/etna/etna_zsa.h
index 260601a..44aca05 100644
--- a/src/gallium/drivers/etna/etna_zsa.h
+++ b/src/gallium/drivers/etna/etna_zsa.h
@@ -28,5 +28,25 @@ struct pipe_context;
 
 void etna_pipe_zsa_init(struct pipe_context *pipe);
 
+#define VIVS_PE_STENCIL_CONFIG                    0x0000141c
+#define VIVS_PE_STENCIL_CONFIG_MODE__MASK         0x00000003
+#define VIVS_PE_STENCIL_CONFIG_MODE__SHIFT        0
+#define VIVS_PE_STENCIL_CONFIG_MODE_DISABLED      0x00000000
+#define VIVS_PE_STENCIL_CONFIG_MODE_ONE_SIDED     0x00000001
+#define VIVS_PE_STENCIL_CONFIG_MODE_TWO_SIDED     0x00000002
+#define VIVS_PE_STENCIL_CONFIG_MODE_MASK          0x00000010
+#define VIVS_PE_STENCIL_CONFIG_REF_FRONT_MASK     0x00000020
+#define VIVS_PE_STENCIL_CONFIG_MASK_FRONT_MASK    0x00000040
+#define VIVS_PE_STENCIL_CONFIG_WRITE_MASK_MASK    0x00000080
+#define VIVS_PE_STENCIL_CONFIG_REF_FRONT__MASK    0x0000ff00
+#define VIVS_PE_STENCIL_CONFIG_REF_FRONT__SHIFT   8
+#define VIVS_PE_STENCIL_CONFIG_REF_FRONT(x)       (((x) << VIVS_PE_STENCIL_CONFIG_REF_FRONT__SHIFT) & VIVS_PE_STENCIL_CONFIG_REF_FRONT__MASK)
+#define VIVS_PE_STENCIL_CONFIG_MASK_FRONT__MASK   0x00ff0000
+#define VIVS_PE_STENCIL_CONFIG_MASK_FRONT__SHIFT  16
+#define VIVS_PE_STENCIL_CONFIG_MASK_FRONT(x)      (((x) << VIVS_PE_STENCIL_CONFIG_MASK_FRONT__SHIFT) & VIVS_PE_STENCIL_CONFIG_MASK_FRONT__MASK)
+#define VIVS_PE_STENCIL_CONFIG_WRITE_MASK__MASK   0xff000000
+#define VIVS_PE_STENCIL_CONFIG_WRITE_MASK__SHIFT  24
+#define VIVS_PE_STENCIL_CONFIG_WRITE_MASK(x)      (((x) << VIVS_PE_STENCIL_CONFIG_WRITE_MASK__SHIFT) & VIVS_PE_STENCIL_CONFIG_WRITE_MASK__MASK)
+
 #endif
 
