diff --git a/gcc/config/mips/mips.h b/gcc/config/mips/mips.h
index 0acce14..c2fc65f 100644
--- a/gcc/config/mips/mips.h
+++ b/gcc/config/mips/mips.h
@@ -1130,6 +1130,7 @@ struct mips_cpu_info {
 %{mmcu} %{mno-mcu} \
 %{msmartmips} %{mno-smartmips} \
 %{mmt} %{mno-mt} \
+%{mmxu} %{mno-mxu} \
 %{mfix-vr4120} %{mfix-vr4130} \
 %{mfix-24k} \
 %{noasmopt:-O0; O0|fno-delayed-branch:-O1; O*:-O2; :-O1} \
diff --git a/gcc/config/mips/mips.opt b/gcc/config/mips/mips.opt
index d8ef2e7..2e47471 100644
--- a/gcc/config/mips/mips.opt
+++ b/gcc/config/mips/mips.opt
@@ -265,6 +265,10 @@ mmt
 Target Report Var(TARGET_MT)
 Allow the use of MT instructions
 
+mmxu
+Target Report Var(TARGET_MXU)
+Allow the use of MXU instructions
+
 mno-float
 Target Report RejectNegative Var(TARGET_NO_FLOAT) Condition(TARGET_SUPPORTS_NO_FLOAT)
 Prevent the use of all floating-point operations
