diff -ur busybox-1.20.2.org/networking/ftpd.c busybox-1.20.2/networking/ftpd.c
--- busybox-1.20.2.org/networking/ftpd.c	2012-06-26 15:35:45.000000000 +0200
+++ busybox-1.20.2/networking/ftpd.c	2013-07-01 23:17:43.000000000 +0200
@@ -707,7 +707,7 @@
 
 	/* -n prevents user/groupname display,
 	 * which can be problematic in chroot */
-	ls_fd = popen_ls((opts & LONG_LISTING) ? "-l" : "-1");
+	ls_fd = popen_ls((opts & LONG_LISTING) ? "-lA" : "-1A");
 	ls_fp = xfdopen_for_read(ls_fd);
 
 	if (opts & USE_CTRL_CONN) {
