diff -ru SDL2_ttf-2.0.12.org/configure sdl2_ttf-2.0.12/configure
--- SDL2_ttf-2.0.12.org/configure	2013-08-10 19:42:03.000000000 +0200
+++ sdl2_ttf-2.0.12/configure	2014-07-03 23:06:17.808621757 +0200
@@ -12957,17 +12957,12 @@
 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 
- #include "SDL_opengl.h"
+ #include "SDL_config.h"
 
-int
-main ()
-{
+ #ifndef SDL_VIDEO_OPENGL
+ #error bla
+ #endif
 
- GLuint texture;
-
-  ;
-  return 0;
-}
 _ACEOF
 if ac_fn_c_try_compile "$LINENO"; then :
 
