--- a/src/SDL12_compat.c	2021-06-11 10:34:18.743740194 +0200
+++ b/src/SDL12_compat.c	2021-06-11 10:34:37.780010611 +0200
@@ -3875,7 +3875,7 @@
         case  8: appfmt = SDL_PIXELFORMAT_INDEX8; break;
         case 16: appfmt = SDL_PIXELFORMAT_RGB565; FIXME("bgr instead of rgb?"); break;
         case 24: appfmt = SDL_PIXELFORMAT_RGB24; FIXME("bgr instead of rgb?"); break;
-        case 32: appfmt = SDL_PIXELFORMAT_XRGB8888; FIXME("bgr instead of rgb?"); break;
+        //case 32: appfmt = SDL_PIXELFORMAT_XRGB8888; FIXME("bgr instead of rgb?"); break;
         default: SDL20_SetError("Unsupported bits-per-pixel"); return NULL;
     }
 
