--- libdrm-2.4.97/configure.ac	2019-01-22 17:32:41.000000000 +0100
+++ libdrm-2.4.97/configure_mod.ac	2020-06-06 02:10:47.427399000 +0200
@@ -91,37 +91,37 @@
 AC_ARG_ENABLE([udev],
               [AS_HELP_STRING([--enable-udev],
                               [Enable support for using udev instead of mknod (default: disabled)])],
-              [UDEV=$enableval], [UDEV=no])
+              [UDEV=$enableval], [UDEV=yes])

 AC_ARG_ENABLE(libkms,
	      AS_HELP_STRING([--disable-libkms],
	      [Disable KMS mm abstraction library (default: auto, enabled on supported platforms)]),
-	      [LIBKMS=$enableval], [LIBKMS=auto])
+		  [LIBKMS=$enableval], [LIBKMS=yes])

 AC_ARG_ENABLE(intel,
 	      AS_HELP_STRING([--disable-intel],
 	      [Enable support for intel's KMS API (default: auto, enabled on x86)]),
-	      [INTEL=$enableval], [INTEL=auto])
+	      [INTEL=$enableval], [INTEL=no])
 
 AC_ARG_ENABLE(radeon,
 	      AS_HELP_STRING([--disable-radeon],
 	      [Enable support for radeon's KMS API (default: auto)]),
-	      [RADEON=$enableval], [RADEON=auto])
+	      [RADEON=$enableval], [RADEON=no])
 
 AC_ARG_ENABLE(amdgpu,
 	      AS_HELP_STRING([--disable-amdgpu],
 	      [Enable support for amdgpu's KMS API (default: auto)]),
-	      [AMDGPU=$enableval], [AMDGPU=auto])
+	      [AMDGPU=$enableval], [AMDGPU=no])
 
 AC_ARG_ENABLE(nouveau,
 	      AS_HELP_STRING([--disable-nouveau],
 	      [Enable support for nouveau's KMS API (default: auto)]),
-	      [NOUVEAU=$enableval], [NOUVEAU=auto])
+	      [NOUVEAU=$enableval], [NOUVEAU=no])
 
 AC_ARG_ENABLE(vmwgfx,
 	      AS_HELP_STRING([--disable-vmwgfx],
 	      [Enable support for vmwgfx's KMS API (default: yes)]),
-	      [VMWGFX=$enableval], [VMWGFX=yes])
+	      [VMWGFX=$enableval], [VMWGFX=no])
 
 AC_ARG_ENABLE(omap-experimental-api,
 	      AS_HELP_STRING([--enable-omap-experimental-api],
@@ -156,7 +156,7 @@
 AC_ARG_ENABLE(etnaviv-experimental-api,
 	      AS_HELP_STRING([--enable-etnaviv-experimental-api],
 	      [Enable support for etnaviv's experimental API (default: disabled)]),
-	      [ETNAVIV=$enableval], [ETNAVIV=no])
+	      [ETNAVIV=$enableval], [ETNAVIV=yes])
 
 AC_ARG_ENABLE(install-test-programs,
 		  AS_HELP_STRING([--enable-install-test-programs],
