diff -ru a/deps/v8/src/objects.h b/deps/v8/src/objects.h
--- a/deps/v8/src/objects.h	2014-07-31 19:24:06.000000000 +0200
+++ b/deps/v8/src/objects.h	2019-09-27 13:54:42.369331143 +0200

@@ -7373,7 +7373,7 @@
   STATIC_CHECK(IS_POWER_OF_TWO(kMaxCachedArrayIndexLength + 1));
 
   static const int kContainsCachedArrayIndexMask =
+      (~static_cast<unsigned>(kMaxCachedArrayIndexLength) << kArrayIndexHashLengthShift) |
-      (~kMaxCachedArrayIndexLength << kArrayIndexHashLengthShift) |
       kIsNotArrayIndexMask;
 
   // Value of empty hash field indicating that the hash is not computed.
