diff -aurp host-nodejs-0.10.30-orig/configure host-nodejs-0.10.30/configure
--- host-nodejs-0.10.30-orig/configure	2019-10-26 23:18:22.770203538 +0800
+++ host-nodejs-0.10.30/configure	2019-10-26 23:19:02.299566018 +0800
@@ -485,7 +485,10 @@ def configure_node(o):
   o['variables']['clang'] = 1 if is_clang else 0
 
   if not is_clang and cc_version != 0:
-    o['variables']['gcc_version'] = 10 * cc_version[0] + cc_version[1]
+    if len(cc_version) > 1:
+      o['variables']['gcc_version'] = 10 * cc_version[0] + cc_version[1]
+    else:
+      o['variables']['gcc_version'] = 10 * cc_version[0]
 
   # clang has always supported -fvisibility=hidden, right?
   if not is_clang and cc_version < (4,0,0):
