diff -ru a/deps/v8/src/liveedit.cc b/deps/v8/src/liveedit.cc
--- a/deps/v8/src/liveedit.cc	2014-07-31 19:24:06.000000000 +0200
+++ b/deps/v8/src/liveedit.cc	2019-09-27 14:16:35.836217070 +0200
@@ -202,7 +202,7 @@
 
   static const int kDirectionSizeBits = 2;
   static const int kDirectionMask = (1 << kDirectionSizeBits) - 1;
-  static const int kEmptyCellValue = -1 << kDirectionSizeBits;
+  static const int kEmptyCellValue = -(1 << kDirectionSizeBits);
 
   // This method only holds static assert statement (unfortunately you cannot
   // place one in class scope).
