diff -ru a/deps/v8/src/objects.h b/deps/v8/src/objects.h
--- a/deps/v8/src/objects.h	2014-07-31 19:24:06.000000000 +0200
+++ b/deps/v8/src/objects.h	2019-09-27 13:54:42.369331143 +0200

@@ -5185,7 +5185,7 @@
   static const int kElementsKindBitCount = 5;
 
   // Derived values from bit field 2
+  static const int kElementsKindMask = -(1 << kElementsKindShift) &
-  static const int kElementsKindMask = (-1 << kElementsKindShift) &
       ((1 << (kElementsKindShift + kElementsKindBitCount)) - 1);
   static const int8_t kMaximumBitField2FastElementValue = static_cast<int8_t>(
       (FAST_ELEMENTS + 1) << Map::kElementsKindShift) - 1;
