--- a/IMG_png.c	2019-06-30 06:52:03.000000000 +0200
+++ b/IMG_png.c	2021-06-06 09:05:44.499550925 +0200
@@ -86,6 +86,7 @@
 #if (PNG_LIBPNG_VER_MINOR < 6)
 typedef png_structp png_structrp;
 typedef png_infop png_inforp;
+typedef png_bytep png_const_bytep;
 typedef png_const_structp png_const_structrp;
 typedef png_const_infop png_const_inforp;
 /* noconst15: version < 1.6 doesn't have const, >= 1.6 adds it */
