diff -ru a/gcc/config/mips/mips.h b/gcc/config/mips/mips.h
--- a/gcc/config/mips/mips.h	2014-03-04 22:39:50.000000000 +0100
+++ b/gcc/config/mips/mips.h	2014-07-03 02:57:47.988146466 +0200
@@ -1178,6 +1178,7 @@
 %{mvirt} %{mno-virt} \
 %{msmartmips} %{mno-smartmips} \
 %{mmt} %{mno-mt} \
+%{mmxu} %{mno-mxu} \
 %{mfix-rm7000} %{mno-fix-rm7000} \
 %{mfix-vr4120} %{mfix-vr4130} \
 %{mfix-24k} \
diff -ru a/gcc/config/mips/mips.opt b/gcc/config/mips/mips.opt
--- a/gcc/config/mips/mips.opt	2014-02-21 14:30:47.000000000 +0100
+++ b/gcc/config/mips/mips.opt	2014-07-03 02:15:50.864894973 +0200
@@ -303,6 +303,10 @@
 Target Report Var(TARGET_MT)
 Allow the use of MT instructions
 
+mmxu
+Target Report Var(TARGET_MXU)
+Allow the use of MXU instructions
+
 mno-float
 Target Report RejectNegative Var(TARGET_NO_FLOAT) Condition(TARGET_SUPPORTS_NO_FLOAT)
 Prevent the use of all floating-point operations
