diff -ru apitrace-5.0.org/dispatch/glproc_egl.cpp apitrace-5.0/dispatch/glproc_egl.cpp
--- apitrace-5.0.org/dispatch/glproc_egl.cpp	2014-05-03 00:22:02.000000000 +0200
+++ apitrace-5.0/dispatch/glproc_egl.cpp	2014-09-02 22:59:32.682791545 +0200
@@ -70,7 +70,10 @@
 void *
 _getPublicProcAddress(const char *procName)
 {
-#if defined(ANDROID)
+/* For some reason, the other path fails to find the function, so follow the Android
+ * path for now.
+ */
+#if 1 //defined(ANDROID)
     /*
      * Android does not support LD_PRELOAD.  It is assumed that applications
      * are explicitely loading egltrace.so.
